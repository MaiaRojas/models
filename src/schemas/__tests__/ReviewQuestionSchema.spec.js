const mongoose = require('mongoose/browser');
const { ReviewQuestionSchema } = require('../')(mongoose);

describe('ReviewQuestionSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, ReviewQuestionSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should fail validation when multipleChoice question is missing options', (done) => {
    const doc = new mongoose.Document({
      i18nId: 'foo',
      type: 'multipleChoice',
      visibility: 'private',
    }, ReviewQuestionSchema);

    doc.validate((err) => {
      expect(err.message).toBe('Options is required when type is multipleChoice');
      done();
    });
  });

  it('should fail validation when question is missing i18nId', () => {
    const doc = new mongoose.Document({
      type: 'multipleChoice',
      visibility: 'private',
      options: ['1', '2'],
    }, ReviewQuestionSchema);

    return doc.validate()
      .catch(err => expect(err.message).toMatchSnapshot());
  });

  it('should fail validation providing an unknown type', () => {
    const doc = new mongoose.Document({
      i18nId: 'bar',
      type: 'other',
      visibility: 'public',
    }, ReviewQuestionSchema);

    return doc.validate()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should validate good review Open Question', (done) => {
    const doc = new mongoose.Document({
      i18nId: 'foo',
      type: 'open',
      visibility: 'public',
    }, ReviewQuestionSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });

  it('should validate good review MultipleChoice Question', (done) => {
    const doc = new mongoose.Document({
      i18nId: 'bar',
      type: 'multipleChoice',
      visibility: 'private',
      options: ['1', '2'],
    }, ReviewQuestionSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
