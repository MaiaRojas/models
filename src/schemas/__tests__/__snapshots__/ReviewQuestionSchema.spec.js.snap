// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewQuestionSchema should fail validation providing an unknown type 1`] = `
Object {
  "type": [ValidatorError: \`other\` is not a valid enum value for path \`type\`.],
}
`;

exports[`ReviewQuestionSchema should fail validation when missing fields are provided 1`] = `
Object {
  "i18nId": [ValidatorError: Path \`i18nId\` is required.],
  "type": [ValidatorError: Path \`type\` is required.],
  "visibility": [ValidatorError: Path \`visibility\` is required.],
}
`;

exports[`ReviewQuestionSchema should fail validation when question is missing i18nId 1`] = `"Validation failed: i18nId: Path \`i18nId\` is required."`;
